### PR TITLE
devtools: replace github-merge with python version

### PR DESCRIPTION
**Pull request by laanwj**
This is meant to be a direct translation of the bash script.

The only major difference is that it retrieves the PR title from github, thus creating pull messages like:

```
Merge pull request #12345: Expose transaction temperature over RPC
```

[skip ci]

### CODE REVIEW

**Comment by jonasschnelli** [comment]
Nice!
Tested ACK (on a different repo: https://github.com/digitalbitbox/dbb-app/commit/a592638e733a124426145cb0601907561ecd9975).

Thought for future additions:
* show the head SHA256 hash from the PR (possible short compare against the one you have ACKed)
* short command to diff the PR (temp bash alias instead of typing `git diff HEAD~1`?)

**Comment by MarcoFalke** [comment]
utACK  b86484c

File: contrib/devtools/github-merge.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    '''
+    Retrieve pull request title from github.
+    Return None if no title can be found, or an error happens.
+    '''
+    import urllib2,json
+    try:
+        req = urllib2.Request("https://api.github.com/repos/"+repo+"/pulls/"+pull)
+        result = urllib2.urlopen(req)
+        result = json.load(result)
+        return result['title']
```
Comment (MarcoFalke) [inline contrib/devtools/github-merge.py]: Could make sense to "sanitize" the title such that it fits in the git(hub) subject line.

That is, `return title[0:min(80, len(title)) - len("Merge pull request #xxxx: "]`

Comment (laanwj) [inline contrib/devtools/github-merge.py]: I don't know - I don't like it cut off.
At least github tends to handle long title lines already by adding ... and continuing on the next line.

Comment (laanwj) [inline contrib/devtools/github-merge.py]: Other, less lossy ways to shorten the text may be:
- leave out the "pull request" completely, as in "Merge #1234: bla"
- abbreviate it as PR
...

Comment (MarcoFalke) [inline contrib/devtools/github-merge.py]: ACK on shorten the text.

Comment (jonasschnelli) [inline contrib/devtools/github-merge.py]: ACK on keeping the full title but removing "pull request".

**Comment by laanwj** [comment]
> show the head SHA256 hash from the PR (possible short compare against the one you have ACKed)

The simplest idea there would be to print the commit message for the merge to the console. It has all context information: the pull #,  the pull title, and the SHA (not 256) hashes for the commits in topological order.
(this can be done in a later pull, not going to do so now)

File: contrib/devtools/github-merge.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                exit(6)
+        else:
+            # Verify the result manually.
+            print("Dropping you on a shell so you can try building/testing the merged source.",file=stderr)
+            print("Run 'git diff HEAD~' to show the changes being merged.",file=stderr)
+            print("Type 'exit' when done.",file=stderr)
+            if os.path.isfile('/etc/debian_version'): # Show pull number on Debian default prompt
+                os.putenv('debian_chroot',pull)
+            subprocess.call([BASH,'-i'])
+            reply = ask_prompt("Type 'm' to accept the merge.")
```
Comment (laanwj) [inline contrib/devtools/github-merge.py]: BTW: is anyone opposed to combining 'accept' and 'sign off' into one step? I may be overlooking something, but I've never completely understood why these are separate, why you would accept a change but then not sign it (which is mandatory).

**Comment by MarcoFalke** [comment]
Tested ACK  da6d18b

File: contrib/devtools/github-merge.py (hunk 2 above)
Comment (sipa) [inline contrib/devtools/github-merge.py]: They were separate at a time when the signing was not mandatory :)

**Pull request merged**